java/org/apache/catalina/filters/RemoteIpFilter.java

```diff
@@ -717,6 +541,11 @@ public StringBuffer getRequestURL() {
      */
     private String remoteIpHeader = "X-Forwarded-For";
 
+    /**
+     * @see #setRequestIdHeader(String)
+     */
+    private String requestIdHeader = "";
```

Comment:
Shouldn't it be `null`?

Reply:
In the current state of the implementation I made it "backwards compatible" in the sense that no header is set.  Yes, we can set it to `null` but empty String does not add a risk of NPE so I'm not sure if using `null` adds value.

I'm happy to change it to null if others think that it's more appropriate.